Modules/Scripted/SegmentStatistics/SegmentStatisticsCalculators/SegmentStatisticsCalculatorBase.py

```diff
@@ -0,0 +1,201 @@
+    self.updateGuiFromParameterNode()
+
+  def getParameterNode(self):
+    return self.parameterNode
+
+  def createDefaultOptionsWidget(self):
+    # create list of checkboxes that allow selection of requested keys
+    self.optionsWidget = qt.QWidget()
+    form = qt.QFormLayout(self.optionsWidget)
+
+    # checkbox to enable/disable calculator
+    self.calculatorCheckbox = qt.QCheckBox(self.name+" calculator enabled")
+    self.calculatorCheckbox.checked = True
+    self.calculatorCheckbox.connect('stateChanged(int)', self.updateParameterNodeFromGui)
+    form.addRow(self.calculatorCheckbox)
+
+    # select all/none/default buttons
+    selectAllNoneFrame = qt.QFrame(self.optionsWidget)
+    selectAllNoneFrame.setLayout(qt.QHBoxLayout())
+    selectAllNoneFrame.layout().setSpacing(0)
+    selectAllNoneFrame.layout().setMargin(0)
+    selectAllNoneFrame.layout().addWidget(qt.QLabel("Select measurements: ",self.optionsWidget))
+    selectAllButton = qt.QPushButton('all',self.optionsWidget)
+    selectAllNoneFrame.layout().addWidget(selectAllButton)
+    selectAllButton.connect('clicked()', self.requestAll)
+    selectNoneButton = qt.QPushButton('none',self.optionsWidget)
+    selectAllNoneFrame.layout().addWidget(selectNoneButton)
+    selectNoneButton.connect('clicked()', self.requestNone)
+    selectDefaultButton = qt.QPushButton('default',self.optionsWidget)
+    selectAllNoneFrame.layout().addWidget(selectDefaultButton)
+    selectDefaultButton.connect('clicked()', self.requestDefault)
+    form.addRow(selectAllNoneFrame)
+
+    # checkboxes for individual keys
+    self.requestedKeysCheckboxes = {}
+    requestedKeys = self.getRequestedKeys()
+    for key in self.keys:
+      label = key
+      tooltip = "key: "+key
+      info = self.getMeasurementInfo(key)
+      if info and ("name" in info or "description" in info):
+        label = info["description"] if "description" in info else info["name"]
+        if "name" in info: tooltip += "\nname: " + str(info["name"])
+        if "description" in info: tooltip += "\ndescription: " + str(info["description"])
+        if "units" in info: tooltip += "\nunits: " + (str(info["units"]) if info["units"] else "n/a")
+      checkbox = qt.QCheckBox(label,self.optionsWidget)
+      checkbox.checked = key in requestedKeys
+      checkbox.setToolTip(tooltip)
+      form.addRow(checkbox)
+      self.requestedKeysCheckboxes[key] = checkbox
+      checkbox.connect('stateChanged(int)', self.updateParameterNodeFromGui)
+
+  def updateGuiFromParameterNode(self, caller=None, event=None):
+    if not self.parameterNode:
+      return
+    isEnabled = self.parameterNode.GetParameter(self.name+'.enabled')!='False'
+    self.calculatorCheckbox.checked = isEnabled
+    for (key, checkbox) in self.requestedKeysCheckboxes.iteritems():
+      parameter = key+'.enabled'
+      value = self.parameterNode.GetParameter(parameter)=='True'
+      if checkbox.checked!=value:
+        checkbox.blockSignals(True)
```

Comment:
Always store return value of blockSignals and restore that value. If you hardcode blockSignals(False) then you may unblock a widget that was already blocked before.